stbt/keyboard.py

```diff
@@ -171,6 +171,19 @@ def enter_text(self, text, page):
             The ``page`` instance that you provide must represent the current
             state of the device-under-test.
 
+        :param bool verify_every_keypress:
+            If True, we will read the selected key after every keypress and
+            assert that it matches the model (``graph``). If False (the
+            default) we will only verify the selected key corresponding to each
+            of the characters in ``text``. For example: to get from Q to D on a
+            qwerty keyboard you need to press KEY_DOWN, KEY_RIGHT, KEY_RIGHT.
+            The default behaviour will only verify that the selected key is D
+            after pressing KEY_RIGHT the last time. This is faster, and closer
+            to the way a human uses the on-screen keyboard.
```

Comment:
It's probably closer to how experienced humans use the on-screen keyboard - like manual testing.  It's how I use it, but other members of my family wait after each press before making the next press.

One way this implementation is not like a human, is that if it were to land on the wrong key the human might be surprised, but they'd just carry on from where they ended up and not worry too much.

Reply:
Good point about non-technical users.

I suppose with the original implementation I was kinda hoping it would work like you describe (carry on from where you ended up) but in reality it was just confusing.